dart/dynamics/InverseKinematics.cpp

```diff
@@ -0,0 +1,1202 @@
+  clearCache();
+}
+
+//==============================================================================
+void InverseKinematics::ErrorMethod::setLinearBounds(
+    const std::pair<Eigen::Vector3d, Eigen::Vector3d>& _bounds)
+{
+  setLinearBounds(_bounds.first, _bounds.second);
+}
+
+//==============================================================================
+std::pair<Eigen::Vector3d, Eigen::Vector3d>
+InverseKinematics::ErrorMethod::getLinearBounds() const
+{
+  return std::pair<Eigen::Vector3d, Eigen::Vector3d>(
+        mProperties.mBounds.first.tail<3>(),
+        mProperties.mBounds.second.tail<3>());
+}
+
+//==============================================================================
+void InverseKinematics::ErrorMethod::setErrorLengthClamp(double _clampSize)
+{
+  mProperties.mErrorLengthClamp = _clampSize;
+  clearCache();
+}
+
+//==============================================================================
+double InverseKinematics::ErrorMethod::getErrorLengthClamp() const
+{
+  return mProperties.mErrorLengthClamp;
+}
+
+//==============================================================================
+void InverseKinematics::ErrorMethod::setErrorWeights(
+    const Eigen::Vector6d& _weights)
+{
+  mProperties.mErrorWeights = _weights;
+  clearCache();
+}
+
+//==============================================================================
+const Eigen::Vector6d& InverseKinematics::ErrorMethod::getErrorWeights() const
+{
+  return mProperties.mErrorWeights;
+}
+
+//==============================================================================
+void InverseKinematics::ErrorMethod::setAngularErrorWeights(
+    const Eigen::Vector3d& _weights)
+{
+  mProperties.mErrorWeights.head<3>() = _weights;
+  clearCache();
+}
+
+//==============================================================================
+Eigen::Vector3d InverseKinematics::ErrorMethod::getAngularErrorWeights() const
+{
+  return mProperties.mErrorWeights.head<3>();
+}
+
+//==============================================================================
+void InverseKinematics::ErrorMethod::setLinearErrorWeights(
+    const Eigen::Vector3d& _weights)
+{
+  mProperties.mErrorWeights.tail<3>() = _weights;
+  clearCache();
+}
+
+//==============================================================================
+Eigen::Vector3d InverseKinematics::ErrorMethod::getLinearErrorWeights() const
+{
+  return mProperties.mErrorWeights.tail<3>();
+}
+
+//==============================================================================
+void InverseKinematics::ErrorMethod::clearCache()
+{
+  // This will force the error to be recomputed the next time computeError is
+  // called
+  mLastConfig.resize(0);
+}
+
+//==============================================================================
+InverseKinematics::TaskSpaceRegion::TaskSpaceRegion(
+    InverseKinematics* _ik,
+    const Properties& _properties,
+    bool _computeFromCenter)
+  : ErrorMethod(_ik, "EulerAngleXYZ", _properties),
```

Comment:
Do you have a special reason to assign *EulerAngleXYZ* rather than *TaskSpaceRegion*?. It tells us the way of converting rotational displacement into a 3-dim vector but not the translation part.